Userland/Utilities/cat.cpp

```diff
@@ -37,13 +41,46 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
 
     TRY(Core::System::pledge("stdio"));
 
+    size_t line_count = 1;
     Array<u8, 32768> buffer;
     for (auto const& file : files) {
         while (!file->is_eof()) {
             auto const buffer_span = TRY(file->read_some(buffer));
-            out("{:s}", buffer_span);
+            if (show_lines) {
+                line_count = output_buffer_with_line_numbers(line_count, buffer_span);
+            } else {
+                out("{:s}", buffer_span);
+            }
         }
+        line_count = 1;
     }
 
     return files.size() != paths.size();
 }
+
+size_t output_buffer_with_line_numbers(size_t line_count, AK::Span<unsigned char> buffer_span)
```

Comment:
`Bytes` is a preferred alias for `AK::Span<unsigned char>`, here you can even use `ReadonlyBytes`